test/georeferencing_t.cpp

```diff
@@ -299,6 +299,58 @@ void GeoreferencingTest::testGridScaleFactor()
 		QCOMPARE(georef.toProjectedCoords(map_coord), sw);
 }
 
+void GeoreferencingTest::testGrivationAffected()
```

Comment:
For some reason, if I check out commit with this test, `georeferencing_t` fails. Are all the necessary infrastructure changes in place for the test to succeed at this point in the patch series?

Reply:
No. Perhaps the Mapper project has a convention that a test should not be introduced into the code until it passes. I have been a little influenced by the idea of "Write the tests before you write the code."

In this case, the test failure demonstrates how the code used by the `GeoreferencingDialog` fails to adjust grivation accurately when the user changes the CRS.

Reply:
The conventions are created on the go. Don't get stressed about the surprises. :)

My comment was motivated by code "bisectability" (git bisect). That means the code must compile after every change (i.e. after every patch in the series must take the code base into a consistent state). I personally extend it to the tests too - i.e. tests must pass after every change. I think that this approach makes bug hunting easier.

The "write test, then code" is certainly beneficial and I use it too. However, the order should IHMO be reversed for the version control system - make the change, then test its validity.

Reply:
I also try to introduce the test first, but explicitly mark the expected failures. This proves the existence of the bug. The bug fix is forced to remove the expected failure, which underlines the desired effect.

Reply:
I will change this to mark the expected failures.